Add custom theme and pattern lab dependency

ben-hamelin opened the pull request:
<!-- INSTRUCTIONS
- Please use a meaningful pull request title which does not include the issue
  key or branch name.
- Please apply meaningful GitHub Labels to your pull request such as: PHP,
  Twig, SCSS, JS, etc.
- Please make sure you've reviewed the "Files changed" tab before opening this
  PR to ensure it includes what you expect (and nothing more) and adheres to
  our coding standards.
- Browser requirements can be found in docs/browser-requirements.md from the
  root of the project.
-->

## Summary
<!-- Include a summary of your changes that expands upon the title. -->
This introduces a custom theme (currently extends from Bartik) that will be the primary theme for the distribution. It adds the composer dependency on the ecms_patternlab repo as well, using the Oomph composer installers extender plugin to place the package in the root of the new theme.

## Metadata
<!-- Please fill out ALL metadata. Use N/A when necessary. -->
| Question | Answer |
|----------|--------|
| Did you use a meaningful pull request title? | yes
| Did you apply meaningful labels to the pull request? | yes
| Did you perform a self review first? | yes
| Documentation reflects changes? | no
| `CHANGELOG` reflects changes? | yes
| Unit/Functional tests reflect changes? | no
| Did you perform browser testing? | yes
| Risk level | Low
| Relevant links | JIRA issue, bug reports, security alerts, etc.
https://thinkoomph.jira.com/browse/RIG-16

pfrilling commented on .gitignore:
```diff
@@ -31,4 +31,7 @@ composer.lock
+.DS_Store
+
+# Pattern Lab package will be pulled in by Composer.
+/ecms_base/themes/custom/ecms/ecms_patternlab
```
Add a newline here.

ben-hamelin replied:
check!

pfrilling commented on ecms_base/themes/custom/ecms/ecms.theme:
```diff
@@ -0,0 +1,9 @@
+<?php
```
I don't think we need this file if it is not doing anything. Otherwise, strict type checking and method return hints should be added.

ben-hamelin replied:
Dropped it

pfrilling commented on scripts/develop.sh:
```diff
@@ -157,6 +157,7 @@ echo " Require ${REPOSITORY_NAME} using lando composer "
 echo "--------------------------------------------------"
 
 $LANDO composer config repositories.${INSTALL_PROFILE_DIRECTORY} '{"type": "path", "url": "/'${INSTALL_PROFILE_DIRECTORY}'", "options": {"symlink": true}}'
+$LANDO composer config repositories.ecms_patternlab '{"type": "package", "package": {"name": "state-of-rhode-island-ecms/ecms_patternlab", "version": "1.0.0", "type": "pattern-lab", "source": {"url": "https://github.com/State-of-Rhode-Island-eCMS/ecms_patternlab", "type": "git", "reference": "master"}}}'
```
I think we need to update the README with this information too. The distribution will need to make this composer addition as well. Even though it is listed as a requirement for the profile, I _believe_ the distribution won't know where to place the file.

ben-hamelin replied:
I am working on a feature branch within the ecms_distribution. Pending results of that work I will update documentation and configuration as needed.

pfrilling reviewed (changes requested):
I think this looks good. Just a few nit-picks and a documentation update.

ben-hamelin commented:
@pfrilling Please note that in addition to your comments, I've added some profile install updates here as well (make custom theme default, added admin toolbar). Plus updated and new documentation.

pfrilling commented on ecms_base/themes/custom/ecms/ecms.info.yml:
```diff
@@ -0,0 +1,28 @@
+name: Rhode Island eCMS Theme
+description: "Drupal installation profile for the State of Rhode Island's eCMS system"
```
Nit pick, but I think this description should change to mention it is a custom Drupal theme for the eCMS system.

ben-hamelin replied:
Ah good catch! Fixed that

pfrilling reviewed (changes requested):
One very minor nit pick. Otherwise looks great.

pfrilling reviewed (approved):
Looks good to me. 🎉

pfrilling commented:
I just had that same problem. Still looks good.
